### DISCUSSION TITLE
Escaping semicolon in ODF

### DISCUSSION THREAD

**Discussion by francescoza6** [GrandOrgue/grandorgue#2044] [Help]
Good morning everyone.

First time I post here, so please excuse me for any non standard linking and reference to source code...

I need to load an ODF I made for a commercial hauptwerk sampleset, and some of the paths to the samples contains semicolon in the folder name.

For example:

samples\noises; tracker action\036-c.wav

While I understand this is non-standard naming, I'd like to create an ODF that works so I can share it without having to tell users to rename the samples folders in order to load it (I have the permission from the sample set producer to share it, of course).

I may be missing something obvious (in that case I apologize in advance) but the problem is, I can't find a way to escape that semicolon so GrandOrgue doesn't interpret "; tracker action\036-c.wav" as a comment.

I had a quick look at the source code, and at

src/core/config/GOConfigFileReader.cpp

at line 121 it seems to be where the comments are removed, and I can't find a way to avoid that...

Any help is very appreciated! Thanks!

**Comment by Psi58** [comment]
Hi, could this possibly be the same sampleset referred to in https://github.com/GrandOrgue/OdfEdit/discussions/16#discussioncomment-11243597?
If so, the solution mentioned in that thread works, but I can't  think of a way to fix the problem without manually renaming the affected folders. Ideally the HW sampleset creator would change the original file to comply with GO syntax rules...

**Comment by larspalo** [comment]
It's a design choice we made in https://github.com/GrandOrgue/grandorgue/pull/1359. If a change is desired, open an issue.

The current solution for GrandOrgue is to re-name folders/directories that contain a semicolon (;).
